internal/rootcoord/garbage_collector.go

```diff
@@ -34,125 +27,84 @@ func newGarbageCollectorCtx(s *Core) *GarbageCollectorCtx {
-		log.Error("failed to drop collection index when recovery", zap.Error(err), zap.String("collection", collMeta.Name), zap.Int64("collection id", collMeta.CollectionID))
-		return
-	}
 
-	if err := c.GcCollectionData(ctx, collMeta, ts); err != nil {
-		log.Error("failed to notify datacoord to gc collection when recovery", zap.Error(err), zap.String("collection", collMeta.Name), zap.Int64("collection id", collMeta.CollectionID))
-		return
-	}
-
-	if err := c.s.meta.RemoveCollection(ctx, collMeta.CollectionID, ts); err != nil {
-		log.Error("failed to remove collection when recovery", zap.Error(err), zap.String("collection", collMeta.Name), zap.Int64("collection id", collMeta.CollectionID))
-	}
+	redo := newBaseRedoTask(c.s.stepExecutor)
+	redo.AddAsyncStep(&releaseCollectionStep{
+		baseStep:     baseStep{core: c.s},
+		collectionID: collMeta.CollectionID,
+	})
+	redo.AddAsyncStep(&dropIndexStep{
+		baseStep: baseStep{core: c.s},
+		collID:   collMeta.CollectionID,
+	})
+	redo.AddAsyncStep(&deleteCollectionDataStep{
+		baseStep: baseStep{core: c.s},
+		coll:     collMeta,
+		ts:       ts,
+	})
+	redo.AddAsyncStep(&removeDmlChannelsStep{
+		baseStep:  baseStep{core: c.s},
+		pChannels: collMeta.PhysicalChannelNames,
+	})
+	redo.AddAsyncStep(&deleteCollectionMetaStep{
+		baseStep:     baseStep{core: c.s},
+		collectionID: collMeta.CollectionID,
+		ts:           ts,
+	})
+
+	// err is ignored since no sync steps will be executed.
+	_ = redo.Execute(context.Background())
 }
 
 func (c *GarbageCollectorCtx) RemoveCreatingCollection(collMeta *model.Collection) {
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
-	defer cancel()
-
-	if err := c.s.broker.UnwatchChannels(ctx, &watchInfo{collectionID: collMeta.CollectionID, vChannels: collMeta.VirtualChannelNames}); err != nil {
-		log.Error("failed to unwatch channels when recovery",
-			zap.Error(err),
-			zap.String("collection", collMeta.Name), zap.Int64("collection id", collMeta.CollectionID),
-			zap.Strings("vchans", collMeta.VirtualChannelNames), zap.Strings("pchans", collMeta.PhysicalChannelNames))
-		return
-	}
-
-	if err := c.s.meta.RemoveCollection(ctx, collMeta.CollectionID, collMeta.CreateTime); err != nil {
-		log.Error("failed to remove collection when recovery", zap.Error(err), zap.String("collection", collMeta.Name), zap.Int64("collection id", collMeta.CollectionID))
-	}
+	redo := newBaseRedoTask(c.s.stepExecutor)
+	redo.AddAsyncStep(&unwatchChannelsStep{
+		baseStep:     baseStep{core: c.s},
+		collectionID: collMeta.CollectionID,
+		channels: collectionChannels{
+			virtualChannels:  collMeta.VirtualChannelNames,
+			physicalChannels: collMeta.PhysicalChannelNames,
+		},
+	})
+	redo.AddAsyncStep(&deleteCollectionMetaStep{
+		baseStep:     baseStep{core: c.s},
+		collectionID: collMeta.CollectionID,
+		ts:           collMeta.CreateTime,
+	})
+	// err is ignored since no sync steps will be executed.
+	_ = redo.Execute(context.Background())
```

Comment:
may be we can log.Info this err ?

Reply:
`redo.Execute` here won't return error, it always return `nil`.